brain_friendly.py

```diff
@@ -85,6 +86,17 @@ def get_brace_matches(program):
     else:
         return brace_pairs
 
+
+def eval_file(filename, index, memory, input=None, output=None):
+    f = open(filename, 'r')
+    program = f.read()
+    f.close()
+    return eval_program(program, index, memory, input, output)
+
+
 if __name__ == '__main__':
-    bf_memory = eval_program(bf_program_string, bf_index, bf_memory)
-    print(bf_memory[:10])
+    filename = "test_programs/hello_world/hello_world_commented.bf"
+    output = io.BytesIO()
+    bf_memory = eval_file(filename, bf_index, bf_memory, output=output)
+    output.seek(0)
+    print(output.getvalue())
```

Comment:
There's no need to mess around with an `io.BytesIO` object, that means you won't see any output until the program finishes running! Instead, write directly to the `stdout` file stream. Since `sys.stdout` in Python 3 is a Unicode stream, I recommend either
```python
if hasattr(sys.stdout, 'buffer'):
    output = sys.stdout.buffer
else:
    output = sys.stdout
```
or, probably better, since it's EAFP,
```python
try:
    output = sys.stdout.buffer
except AttributeError:
    output = sys.stdout
```
Both of these will write directly to the output stream, which will make it more responsive, and also conceptually simpler.